do not use deprecated reinit() in trilinos_vector.h:721

```
include/deal.II/lac/trilinos_vector.h:721:7: warning: 'reinit' is deprecated [-Wdeprecated-declarations]
      reinit (vector_partitioner(), v);
      ^
```

happens for example for

```
TrilinosWrappers::MPI::Vector t_src;
Vector<double> tmp;
...
t_src = tmp;
```

--- a/include/deal.II/lac/trilinos_vector.h
+++ b/include/deal.II/lac/trilinos_vector.h
@@ -718,7 +718,14 @@ namespace TrilinosWrappers
                                             )));
         }
 
-      reinit (vector_partitioner(), v);
+      const Epetra_Map &map = vector_partitioner();
+      const int size = map.NumMyElements();
+
+      // Need to copy out values, since the deal.II might not use doubles, so
+      // that a direct access is not possible.
+      for (int i=0; i<size; ++i)
+        (*vector)[0][i] = v(gid(map,i));
+
       return *this;
     }
 

